esmvaltool/diag_scripts/primavera/eady_growth_rate/eady_growth_rate.py

```diff
@@ -0,0 +1,335 @@
+        -------
+        egr: da.array
+            Array containing Eady Growth Rate.
+        """
+        dudz = self.vertical_integration(eastward_wind, geopotential)
+        egr = self.con * abs(fcor) * abs(dudz) / brunt
+
+        return egr
+
+    def seasonal_plots(self, egr, alias):
+        """
+        Plot seasonal Eady Growth rate values.
+        Parameters
+        ----------
+        egr: iris.cube.Cube
+            Cube containing variable egr.
+        alias: str
+            Alias of the dataset.
+        """
+        try:
+            levels = self.cfg['plot_levels']
+        except KeyError:
+            logger.info("Parameter plot_levels is not set in the recipe."
+                        "Plotting all pressure levels instead.")
+            levels = egr.coord('air_pressure').points
+        for level in levels:
+            cube = extract_levels(egr, level, scheme='linear')
+            crs_latlon = ccrs.PlateCarree()
+            axes = plt.axes(projection=ccrs.PlateCarree())
+            axes.coastlines(linewidth=1, color='black')
+            # North Atlantic
+            axes.set_extent((-90.0, 30.0, 20.0, 80.0), crs=crs_latlon)
+            axes.set_yticks(np.linspace(25, 75, 6))
+            qplt.contourf(cube, levels=np.arange(0, np.max(cube.data), 0.05))
+            extension = self.cfg['output_file_type']
+            diagnostic = self.cfg['script']
+            plotname = '_'.join([alias, diagnostic,
+                                 str(int(level))]) + f'.{extension}'
+            plt.savefig(os.path.join(
+                self.cfg[shared_names.PLOT_DIR], plotname))
+            plt.close()
+
+    def save(self, egr, alias, data):
+        """
+        Save results and write provenance.
+        """
+        script = self.cfg[shared_names.SCRIPT]
+        info = data[alias][0]
+        keys = [
+            str(info[key]) for key in ('project', 'dataset', 'exp', 'ensemble',
+                                       'diagnostic', 'start_year', 'end_year')
+            if key in info]
+        output_name = '_'.join(keys) + '.nc'
+        output_file = os.path.join(
+            self.cfg[shared_names.WORK_DIR], output_name)
+        iris.save(egr, output_file)
+
+        caption = ("{script} between {start} and {end}"
+                   "according to {dataset}").format(
+                       script=script.replace(" ", '_'),
+                       start=info['start_year'],
+                       end=info['end_year'],
+                       dataset=info['dataset'])
+        ancestors = []
+        for i in range(len(data[alias])):
+            ancestors.append(data[alias][i]['filename'])
+        record = {
+            'caption': caption,
+            'domains': ['global'],
+            'authors': ['sanchez-gomez_emilia', 'moreno-chamarro_eduardo'],
+            'references': ['acknow_project'],
+            'ancestors': ancestors
+        }
+        with ProvenanceLogger(self.cfg) as provenance_logger:
```

Comment:
I realized you had specific a question about provenance on gitter that I did missed the first time around. I just want to say that this looks good to me. 👍

The references part `'acknow_project'` is basically a placeholder as far as I can tell, unless you have any references that this work is based on/described by. You can link your publication there by adding a bibtex file to the `esmvaltool/references` directory.

Reply:
I don't have a reference as of right now. I'll add it later on if that changes though.